internal/deb/extract_test.go

```diff
@@ -352,6 +352,62 @@ var extractTests = []extractTest{{
 		},
 	},
 	error: `cannot extract from package "test-package": path /dir/ requested twice with diverging mode: 0777 != 0000`,
+}, {
+	summary: "Dangling hard link",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		testutil.Hln(0644, "./link", "./non-existing-target"),
+	}),
+	options: deb.ExtractOptions{
+		Extract: map[string][]deb.ExtractInfo{
+			"/**": []deb.ExtractInfo{{
+				Path: "/**",
+			}},
+		},
+	},
+	error: `cannot extract from package "test-package": no content at \/non-existing-target`,
+}, {
+	summary: "Hard link to symlink does not follow symlink",
```

Comment:
@letFunny I remember the naming of this test is suggested by you. Do you think this suggestion describes our test expectations better?

Reply:
I would keep it as is, I think it describes the behavior better, I find that `produces a symlink` can be more confusing because the symlink can point to a symlink or things like that. Happy to discuss it further.